web/src/buddy_list.ts

```diff
@@ -767,13 +767,13 @@ export class BuddyList extends BuddyListConf {
 
     // From `type List<Key>`, where the key is a user_id.
     first_key(): number | undefined {
-        if (this.participant_user_ids.length > 0) {
+        if (this.participant_user_ids.length > 0 && !this.participants_is_collapsed) {
             return this.participant_user_ids[0];
```

Comment:
Reading this code, probably eventually it will want a refactor where the sections are objects and we can make this a loop over the sections, check something like `!section.is_collapsed && section.user_ids.length > 0`, for example.

(I think it's more readable to check if the section is collapsed first).

Reply:
Oh yeah I like that a lot, thanks for suggesting it. Not too hard to just do now!